### DISCUSSION TITLE
API Gateway Lambda Proxy Integration with RESPONSE_STREAM

### DISCUSSION THREAD

**Discussion by maxwellyc** [awslabs/aws-lambda-web-adapter#513] [Q&A] [answered]
I'm getting Lambda malformed error when I use Lambda Proxy Integration in API Gateway, with a FastAPI/Gradio Lambda function deployed using LWA.
It works somewhat with AWS_LWA_INVOKE_MODE=BUFFERED but I'd really like to use RESPONSE_STREAM instead. The issue is the response doesn't have "body" all the time since it's a streaming response, any suggestions?
The Lambda Proxy Integration is mandatory if I give /{proxy+} resource in the API, since I want to use a wild card resource path, that's the only option in the web console.

**Comment by bnusunny** [comment, marked as answer]
API Gateway does not support response streaming at the moment. You need to use Lambda Function URL or another proxy layer to invoke Lambda API directly. 

https://docs.aws.amazon.com/lambda/latest/dg/configuration-response-streaming.html

**Reply by maxwellyc** [reply]
Thank you for confirming this.
I have a question that's not this thread/discussion-board, but if you could point me on the right track I'd really appreciate it:
The issue I have right now with Function URL is the AWS_IAM auth type complaining about invalid signature. 
I have a CloudFront distribution on top of the Lambda Function URL, and I have OAC set up and it works except for POST requests. Because my company requires Okta, we have to modify the "host" in the header within the web app running on LWA so that Okta doesn't see the Function URL as the redirect, so my current assumption is that the signature error is because the header was modified, giving 404 error codes. Do you know if this could be the reason? 
What I'm not sure about my assumption is that in the AWS_IAM auth type documentation: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-lambda.html 
says "_your users must include the payload hash value_", I know the OAC is already signing the request, and I didn't modify the payload (I'm assuming it just means the body of the response?), but just the header, so why would this still break? Or maybe "payload" here also means the header? (Sorry, new to web app concepts and might be due to my misunderstanding).

**Reply by bnusunny** [reply]
Lambda Function URL relies on its host name. The request will fail if you change the HOST. You should keep the HOST intacted and handle it in your application.

**Comment by bnusunny** [comment]
If you want to deploy a proxy layer, checkout this project. https://github.com/aws-samples/aws-lambda-web-gateway

Deploy it to App Runner, you will have a serverless gateway that supports response streaming from Lambda.

**Comment by constja** [comment]
Hello, is this question still up to date in context of https://aws.amazon.com/blogs/compute/building-responsive-apis-with-amazon-api-gateway-response-streaming/

**Reply by bnusunny** [reply]
Yes, with this new launch, we can use API Gateway with Lambda for response streaming.

**Reply by constja** [reply]
just implemented a reference hosting a caddy in a lambda yesterday. works great. 
Also noteworthy is, that the streaming has a 50mb limit, but it appears that this is a soft limit which can be increased with a support ticket
